src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -82,6 +104,9 @@ def sagemaker_train(train_config, data_config, train_path, val_path, model_dir,
     :param sm_current_host:
     :param checkpoint_config:
     """
+    hpo_early_stopping = train_config.pop("hpo_early_stopping", False)
+    hpo_early_stopping = True if hpo_early_stopping == "true" else False
```

Comment:
- There were two options for the workflow. One was to remove it before validation and set it to my own variable, and the other was to validate it and set it in `train_job` function. The reason I went with the first was because this parameter is not going to be called in `xgboost.train`, therefore, I wanted to keep it separate. However, based on the comment below, if I were to validate it in Hyperparameter_validation then I can move these checks in the `train_job` function rather than `sagemaker_train`. And, this will not require an additional parameter in the `train_job` function.
- Currently, the defined use case is only for HPO. To give it a more general name, we can name it `early_stopping_support`(based on general use case) or `save_intermediate_model`(based on functionality)?